### PR TITLE
net: lib: fota_download: Add option to define host port number

### PR DESCRIPTION
**Pull request by tiagosmc**
The download client already supports connecting to a port number other than the default HTTP/HTTPS port. However, this functionality was not exposed in `fota_download_start`. This adds the functionality of defining the host port number when trying to download a file with `fota_download` in case you are not using the typical 80 or 443 ports.

Signed-off-by: Tiago Costa <tiagosmcosta@gmail.com>

### CODE REVIEW

File: include/net/fota_download.h [hunk 1]
```diff
@@ -76,12 +76,14 @@ int fota_download_init(fota_download_callback_t client_callback);
...
  * @param sec_tag Security tag you want to use with HTTPS set to -1 to Disable.
+ * @param port Port which to connect to. Set to 0 for default HTTP/HTTPS port.
  *
  * @retval 0	     If download has started successfully.
  * @retval -EALREADY If download is already ongoing.
  *                   Otherwise, a negative value is returned.
  */
-int fota_download_start(const char *host, const char *file, int sec_tag);
+int fota_download_start(const char *host, const char *file, int sec_tag,
+			int port);
```
Comment (lemrey) [inline include/net/fota_download.h:86]: Could you let `port` be an `u16_t` instead? that way it's the same type as the field in `download_client_cfg`.

Comment (tiagosmc) [reply include/net/fota_download.h]: Ah, yes, of course! Thanks.

**Comment by tiagosmc** [comment]
I can't figure out why this is failing: **CI/Jenkins/NRF/TEST_NRF/TEST_BLE**. Can someone please tell me why this is failing and if I've missed anything? Thanks

**Comment by rlubos** [comment]
@lemrey, are we good?

**Comment by lemrey** [comment]
Updated changelog.

**Pull request merged**